src/app/components/audio-recordings/pages/download/download.component.ts

```diff
@@ -0,0 +1,389 @@
+      }
+    }
+
+    if (this.region) {
+      this.fields.push(this.getRegionFormInput(hideProjects));
+    } else {
+      // TODO Future work, read regions/sites list and make typeahead
+    }
+
+    if (this.site) {
+      this.fields.push(this.getSiteFormInput(hideProjects));
+    } else {
+      // TODO Future work, read sites list and make typeahead
+    }
+
+    this.fields.push(...this.dateSelectionRange());
+  }
+
+  public get project(): Project {
+    return this.models[projectKey] as Project;
+  }
+  public get region(): Region {
+    return this.models[regionKey] as Region;
+  }
+  public get site(): Site {
+    return this.models[siteKey] as Site;
+  }
+
+  private getProjectFormInput(): FormlyFieldConfig {
+    return {
+      key: "projectId",
+      type: "select",
+      defaultValue: this.project.id,
+      templateOptions: {
+        label: "Project",
+        // TODO Allow user to change selected site
+        disabled: true,
+        options: [{ value: this.project.id, label: this.project.name }],
+      },
+    };
+  }
+
+  private getRegionFormInput(hideProject: boolean): FormlyFieldConfig {
+    return {
+      key: "regionId",
+      type: "select",
+      defaultValue: this.region.id,
+      templateOptions: {
+        label: hideProject ? "Project" : "Site",
+        // TODO Allow user to change selected site
+        disabled: true,
+        options: [{ value: this.region.id, label: this.region.name }],
+      },
+    };
+  }
+
+  private getSiteFormInput(hideProject: boolean): FormlyFieldConfig {
+    return {
+      key: "siteId",
+      type: "select",
+      defaultValue: this.site.id,
+      templateOptions: {
+        label: hideProject ? "Site" : "Point",
+        // TODO Allow user to change selected site
+        disabled: true,
+        options: [{ value: this.site.id, label: this.site.name }],
+      },
+    };
+  }
+
+  private dateSelectionRange(): FormlyFieldConfig[] {
+    return [
+      {
+        key: "recordingStartedBefore",
+        type: "input",
+        templateOptions: {
+          label: "Date recording started before",
+          required: false,
+          type: "datetime-local",
+          description:
+            "The UTC (+00:00) time zone is used for all date filtering. " +
+            "If you're unsure if your data will be included in a filter, expand your date range by one-day in either direction.",
+        },
+      },
+      {
+        key: "recordingFinishedBefore",
+        type: "input",
+        templateOptions: {
+          label: "Date recording ended before",
+          required: false,
+          type: "datetime-local",
+          description:
+            "The UTC (+00:00) time zone is used for all date filtering. " +
+            "If you're unsure if your data will be included in a filter, expand your date range by one-day in either direction.",
+        },
+      },
+    ];
+  }
+
+  public updateHref(): void {
+    if (!this.project) {
+      return;
+    }
```

Comment:
this doesn't make sense from what i've seen so far? why is only project tested for presence?

Reply:
A project must exist for the script to be of any use, would you prefer for it to just return a script with no filled in values if project is not defined?

Reply:
Why must a project exist?

We need zero to many of projects, regions, sites. If we had 0 of all of them we'd be effectively downloading all recordings